nats.go

```diff
@@ -1283,6 +1285,24 @@ func (nc *Conn) sendConnect() error {
 	return nil
 }
 
+// read byte-by-byte from the connection until the given 'c' character is found.
+func (nc *Conn) readStringUpTo(c byte) (string, error) {
```

Comment:
I think their is a func in the stdlib that does this no?

Reply:
I could not find. We cannot use a buffered reader because it does fill up the buffer in the background, thing that we don't want to happen. I looked at a buffered reader with `io.LimitedReader`, but then you need to know the number of bytes you want to read.

I did rename the function to `readProto` and don't pass the delimiter anymore. This function is not meant to be an all purpose function.

Reply:
ok